src/connector.rs

```diff
@@ -245,29 +245,29 @@ mod tests {
         assert_eq!(message, "unsupported scheme http");
     }
 
-    fn tls_config() -> rustls::ClientConfig {
-        #[cfg(feature = "rustls-platform-verifier")]
-        return rustls::ClientConfig::builder()
-            .with_platform_verifier()
-            .with_no_client_auth();
-
-        #[cfg(feature = "rustls-native-certs")]
-        return rustls::ClientConfig::builder()
-            .with_native_roots()
-            .unwrap()
-            .with_no_client_auth();
-
-        #[cfg(feature = "webpki-roots")]
-        return rustls::ClientConfig::builder()
-            .with_webpki_roots()
-            .with_no_client_auth();
-    }
-
     async fn connect(
         https_only: bool,
         https: bool,
     ) -> Result<MaybeHttpsStream<TokioIo<TcpStream>>, BoxError> {
-        let builder = HttpsConnectorBuilder::new().with_tls_config(tls_config());
+        let config_builder = rustls::ClientConfig::builder();
```

Comment:
consider cfg_if? unused code warnings aside, this is a lot less readable than before, to my eye

Reply:
Good idea -- how about now?